exporter/opentelemetry-exporter-otlp/src/opentelemetry/exporter/otlp/metrics_exporter/__init__.py

```diff
@@ -0,0 +1,191 @@
+    OTLPExporterMixin,
+    _get_resource_data,
+)
+from opentelemetry.proto.collector.metrics.v1.metrics_service_pb2 import (
+    ExportMetricsServiceRequest,
+)
+from opentelemetry.proto.collector.metrics.v1.metrics_service_pb2_grpc import (
+    MetricsServiceStub,
+)
+from opentelemetry.proto.common.v1.common_pb2 import StringKeyValue
+from opentelemetry.proto.metrics.v1.metrics_pb2 import (
+    DoubleDataPoint,
+    InstrumentationLibraryMetrics,
+    Int64DataPoint,
+)
+from opentelemetry.proto.metrics.v1.metrics_pb2 import (
+    Metric as CollectorMetric,
+)
+from opentelemetry.proto.metrics.v1.metrics_pb2 import (
+    MetricDescriptor,
+    ResourceMetrics,
+)
+from opentelemetry.sdk.metrics import (
+    Counter,
+    SumObserver,
+    UpDownCounter,
+    UpDownSumObserver,
+    ValueObserver,
+    ValueRecorder,
+)
+from opentelemetry.sdk.metrics.export import (
+    MetricsExporter,
+    MetricsExportResult,
+)
+
+logger = logging.getLogger(__name__)
+
+
+def _get_data_points(sdk_metric, data_point_class):
+
+    data_points = []
+
+    for (
+        label,
+        bound_counter,
+    ) in sdk_metric.instrument.bound_instruments.items():
+
+        string_key_values = []
+
+        for label_key, label_value in label:
+            string_key_values.append(
+                StringKeyValue(key=label_key, value=label_value)
+            )
+
+        for view_data in bound_counter.view_datas:
+
+            if view_data.labels == label:
+
+                data_points.append(
+                    data_point_class(
+                        labels=string_key_values,
+                        value=view_data.aggregator.current,
+                    )
+                )
+                break
+
+    return data_points
+
+
+def _get_temporality(instrument):
+    # pylint: disable=no-member
+    if isinstance(instrument, (Counter, UpDownCounter)):
+        temporality = MetricDescriptor.Temporality.DELTA
+    elif isinstance(instrument, (ValueRecorder, ValueObserver)):
+        temporality = MetricDescriptor.Temporality.INSTANTANEOUS
+    elif isinstance(instrument, (SumObserver, UpDownSumObserver)):
+        temporality = MetricDescriptor.Temporality.CUMULATIVE
+    else:
+        raise Exception(
+            "No temporality defined for instrument type {}".format(
+                type(instrument)
+            )
+        )
+
+    return temporality
+
+
+def _get_type(value_type):
+    # pylint: disable=no-member
+    if value_type is int:
+        type_ = MetricDescriptor.Type.INT64
+
+    elif value_type is float:
+        type_ = MetricDescriptor.Type.DOUBLE
+
+    # FIXME What are the types that correspond with
+    # MetricDescriptor.Type.HISTOGRAM and
+    # MetricDescriptor.Type.SUMMARY?
+    else:
+        raise Exception(
+            "No type defined for valie type {}".format(type(value_type))
+        )
+
+    return type_
+
+
+class OTLPMetricsExporter(MetricsExporter, OTLPExporterMixin):
```

Comment:
Doesn't look like this class actually implements `export()`? I checked the mixin as well, is it planned for another PR?

Reply:
Fixed